exchanges/okx/okx_wrapper.go

```diff
@@ -131,27 +131,27 @@ func (ok *Okx) SetDefaults() {
 			AutoPairUpdates: true,
 			Kline: kline.ExchangeCapabilitiesEnabled{
 				Intervals: kline.DeployExchangeIntervals(
-					kline.OneMin,
-					kline.ThreeMin,
-					kline.FiveMin,
-					kline.FifteenMin,
-					kline.ThirtyMin,
-					kline.OneHour,
-					kline.TwoHour,
-					kline.FourHour,
-					kline.SixHour,
-					kline.TwelveHour,
-					kline.OneDay,
-					kline.TwoDay,
-					kline.ThreeDay,
-					kline.FiveDay,
-					kline.OneWeek,
-					kline.OneMonth,
-					kline.ThreeMonth,
-					kline.SixMonth,
-					kline.OneYear,
+					kline.IntervalCapacity{Interval: kline.OneMin},
+					kline.IntervalCapacity{Interval: kline.ThreeMin},
+					kline.IntervalCapacity{Interval: kline.FiveMin},
+					kline.IntervalCapacity{Interval: kline.FifteenMin},
+					kline.IntervalCapacity{Interval: kline.ThirtyMin},
+					kline.IntervalCapacity{Interval: kline.OneHour},
+					kline.IntervalCapacity{Interval: kline.TwoHour},
+					kline.IntervalCapacity{Interval: kline.FourHour},
+					kline.IntervalCapacity{Interval: kline.SixHour},
+					kline.IntervalCapacity{Interval: kline.TwelveHour},
+					kline.IntervalCapacity{Interval: kline.OneDay},
+					kline.IntervalCapacity{Interval: kline.TwoDay},
+					kline.IntervalCapacity{Interval: kline.ThreeDay},
+					kline.IntervalCapacity{Interval: kline.FiveDay},
+					kline.IntervalCapacity{Interval: kline.OneWeek},
+					kline.IntervalCapacity{Interval: kline.OneMonth},
+					kline.IntervalCapacity{Interval: kline.ThreeMonth},
+					kline.IntervalCapacity{Interval: kline.SixMonth},
+					kline.IntervalCapacity{Interval: kline.OneYear},
 				),
-				ResultLimit: 300,
+				GlobalResultLimit: 300,
```

Comment:
In some fun testing fashion, the function you use for `GetHistoricCandles` is actually `market/history-candles` which has a limit of 100, leading to many empty padded candles in the response:
https://www.okx.com/docs-v5/en/#rest-api-market-data-get-candlesticks-history